packages/api/lib/rulesHelpers.js

```diff
@@ -594,10 +602,8 @@ async function updateRuleTrigger(original, updates, knex) {
       if (enabled && stateChanged && mergedRule.rule.arn) {
         throw new Error('Including rule.arn is not allowed when enabling a disabled rule');
       }
+
       let snsSubscriptionArn;
-      if (mergedRule.rule.arn) {
-        await deleteSnsTrigger(knex, mergedRule);
-      }
```

Comment:
TODO: Not sure about this. This call to `deleteSnsTrigger` does NOT exist on the `Rule.updateRuleTrigger` method but does here.